src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -520,66 +522,195 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
-                    for(const auto &fs : new_faces){
-                        const auto f_A = static_cast<uint64_t>(fs[0] + old_face_count);
-                        const auto f_B = static_cast<uint64_t>(fs[1] + old_face_count);
-                        const auto f_C = static_cast<uint64_t>(fs[2] + old_face_count);
-                        amesh.faces.emplace_back( std::vector<uint64_t>{{f_A, f_B, f_C}} );
+
+                    //routine for hollow structures with an inner contour and an outer contour on both planes
+                    if((N_upper == 2) && (N_lower == 2)){
+                        //check if both planes have enclosed contours
+                        if (contours_are_enclosed(*m_cp_it, pcs.upper.front(), pcs.upper.back())
+                            && contours_are_enclosed(*l_cp_it, pcs.lower.front(), pcs.lower.back())){
+                            //check areas to determine which the inner and outer contours are on both planes
+
+                            //assume 2nd contour is the inner to begin with
+                            auto upper_inner = pcs.upper.back();
+                            auto upper_outer = pcs.upper.front();
+
+                            auto contour = std::begin(pcs.upper);
+                            auto c1_area = std::abs(contour->get().Get_Signed_Area());
+                            ++contour;
+                            auto c2_area = std::abs(contour->get().Get_Signed_Area());
+
+                            if (c1_area < c2_area){
+                                upper_inner = pcs.upper.front();
+                                upper_outer = pcs.upper.back();
+                            }
+
+                            //do the same for the lower plane
+                            auto lower_inner = pcs.lower.back();
+                            auto lower_outer = pcs.lower.front();
+
+                            contour = std::begin(pcs.lower);
+                            c1_area = std::abs(contour->get().Get_Signed_Area());
+                            ++contour;
+                            c2_area = std::abs(contour->get().Get_Signed_Area());
+
+                            if (c1_area < c2_area){
+                                lower_inner = pcs.lower.front();
+                                lower_outer = pcs.lower.back();
+                            }
+
+                            //connect inner contours together and outer contours together
+                            auto new_faces = Estimate_Contour_Correspondence(lower_inner, upper_inner);
+                            add_faces_to_mesh(lower_inner, upper_inner, new_faces);
+                            
+                            new_faces = Estimate_Contour_Correspondence(lower_outer, upper_outer);
+                            add_faces_to_mesh(lower_outer, upper_outer, new_faces);
+                            //move to next iteration of for loop since we have tiled it
+                            continue;
```

Comment:
logic seems weird here, I think if it is 2 to 2 but not nested, will never go to N to N?